### PR TITLE
Downsize Redisson in housekeeping

### PR DESCRIPTION
**Pull request by MocanaAtBroad**
## Context
One clear thing after staring at housekeeping dumps is that we have A LOT of threads and we create more while we operate.
Redisson keeps a lot of threads. And given our usage pattern in housekeeping (i.e., not lots of concurrent requests) it seems unnecessarily heavy. 
So going to try to cut down on threads for Redis configuration.
Most of the work on this PR is actually trying to create a different configuration for Housekeeping. (At some point we should reconsider how we generate our config files: 

**a bash script that calls a Ruby script that calls a Docker image that contains a program that transforms a GO template into a YAML file . Most of the effort in this PR was trying to get the string "housekeeping" through that config file generation pipeline!**

## Checklist

- [X ] I have labeled the type of changes involved using the `C-*` labels.
- [ X] I have assessed potential risks and labeled using the `R-*` labels.
- [X ] I have considered error handling and alerts, and added `L-*` labels as needed.
- [X ] I have considered security and privacy, and added `I-*` labels as needed
- [ X] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [ X] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

Not a demoable thing. Either it works or does not! But if we start seeing timeout errors, then this would indication that we went too far.

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [X ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [X ] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -147,13 +147,19 @@ commands:
...
       - run:
           name: show environment variables
           command: env
       - run:
           name: api-build.sh
-          command: ./api-build.sh v1 ${ENVIRONMENT:-dev} . --gae-config
+          command: |
+            echo "We are generating api-build.sh for application << parameters.application >>"
+            ./api-build.sh v1 ${ENVIRONMENT:-dev} . << parameters.application >> --gae-config
           environment:
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:163]: Additional argument needed

File: .circleci/config.yml [hunk 2]
```diff
@@ -287,6 +297,7 @@ commands:
             cp "$DEPLOY_CONFIG_FILE_DIR/<<parameters.jar_base_name>>.yaml" .
             cp ../output-config/application.conf .
             cp ../output-config/redisson-jcache.yaml .
+            cp ../output-config/redisson-jcache-housekeeping.yaml .
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:300]: The separate housekeeping redisson config file.

File: pepper-apis/api-build.sh [hunk 3]
```diff
@@ -298,6 +303,12 @@ GIT_SHA_SHORT=${GIT_SHA:0:12}
 shift
 shift
 shift
+# Sneaking in another optional config parameter here
+if grep -q -v "\-\-" <<< "$1"; then
+  APPLICATION=$1
+  shift
+fi
+
```
Comment (MocanaAtBroad) [inline pepper-apis/api-build.sh:311]: A little hacky but I am hoping we can streamline the script at some point. A lot of the functionality here is no longer relevant.

File: pepper-apis/config/application.conf.ctmpl [hunk 4]
```diff
@@ -98,7 +99,11 @@
         # Batch size to use for checking status and scheduling
         "batchSize": {{$conf.Data.kits.batchSize}}
     },
+    {{if eq $app "housekeeping"}}
+    "jcacheConfigurationFile": "redisson-jcache-housekeeping.yaml",
+    {{else}}
     "jcacheConfigurationFile": "redisson-jcache.yaml",
+    {{end}}
     "redisServerAddress": "redis://{{$conf.Data.redis.host}}:{{$conf.Data.redis.port}}",
```
Comment (MocanaAtBroad) [inline pepper-apis/config/application.conf.ctmpl:107]: I think this is first time we have a condition in config for housekeeping vs pepper-backend. Thought of generating a separe housekeeping config file but there would be lot of duplication.

File: pepper-apis/config/redisson-jcache-housekeeping.yaml.ctmpl [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  address: "redis://{{$conf.Data.redis.host}}:{{$conf.Data.redis.port}}"
+  subscriptionConnectionMinimumIdleSize: 1
+  subscriptionConnectionPoolSize: 50
+  connectionMinimumIdleSize: 24
+  connectionPoolSize: 64
+  database: 0
+  dnsMonitoringInterval: 5000
+  pingConnectionInterval: 15000
+threads: 16
+nettyThreads: 32
```
Comment (MocanaAtBroad) [inline pepper-apis/config/redisson-jcache-housekeeping.yaml.ctmpl:23]: Went down from 32 to 16 and from 128 to 32. These new values are the default ones.

**Review by yufengwng — APPROVED** [review]

**Pull request merged**